workshops/boost-research-reproducibility-binder/workshop-presentations/zero-to-binder-r.md

```diff
@@ -0,0 +1,308 @@
+- [6. Sharing your Work](#6-sharing-your-work)
+- [7. Accessing data in your Binder](#7-accessing-data-in-your-binder)
+  - [Small public files](#small-public-files)
+  - [Medium public files](#medium-public-files)
+  - [Large public files](#large-public-files)
+  - [Private files](#private-files)
+- [8. Get data with `postBuild`](#8-get-data-with-postbuild)
+- [Beyond Notebooks...](#beyond-notebooks)
+- [Now over to you!](#now-over-to-you)
+
+---
+
+## Running Code is more complicated than Displaying Code
+
+GitHub is a great service for sharing code, but the contents are **static**.
+
+How could you _run_ a GitHub repository **without installing complicated requirements**?
+Or even **in your browser**?
+
+To run code, you need:
+
+- Hardware on which to run the code
+- Software, including:
+  - The code itself
+  - The programming language (e.g. Python, R)
+  - Relevant packages (e.g. pandas, matplotlib, tidyverse, ggplot)
+
+## What Binder Provides
+
+[Binder](https://mybinder.org) is a service that provides your code and the hardware and software to execute it.
+
+You can create a link to a **live, interactive** version of your code!
+
+- An example binder link:
+  > **<https://mybinder.org/v2/gh/trekhleb/homemade-machine-learning/master?filepath=notebooks%2Fanomaly_detection%2Fanomaly_detection_gaussian_demo.ipynb>**
+  From this repo: **<https://github.com/trekhleb/homemade-machine-learning>**
+  - Notice that the Binder link has a similar structure to the GitHub repo link (<github-username>/<github-repo-name>)
+  - The "filepath" argument opens a specific notebook (`.ipynb` file) in the repo
+
+> In R, we can use package [`holepunch`](https://github.com/karthik/holepunch) to **binderise** our projects! The package provides functionality to create the necessary files and automatically configure them with the required information. All files created with `holepunch` are store in a hidden `.binder/` directory and Binder knows to use those files to build your environment.
+
+## 1. Creating a repo to Binderize
+
+**TO DO:** :vertical_traffic_light:
+
+1) Create a new repo on GitHub called "my-first-binder".
+   - Make sure the repository is **public**, _not private_!
+   - Don't forget to initialise with a README!
+2) Create a file called `hello.R` via the web interface with `print("Hello from Binder!")` on the first line and commit to master.
+3) Create a file called `runtime.txt` with `r-YYYY-MM-DD` on the first line, where `YYYY-MM-DD` is today's date (eg `r-2020-05-26`). Commit to master.
+    + _In R you can use `holepunch::write_runtime()` to create a `runtime.txt` in the `.binder/` directory, configured with today's date._
+
+### Why did the repo have to be public?
+
+<mybinder.org> cannot access private repositories as this would require a secret token.
+The Binder team choose not to take on the responsibility of handling secret tokens as <mybinder.org> is a public service and proof of technological concept.
```

Comment:
```suggestion
mybinder.org cannot access private repositories as this would require a secret token.
The Binder team choose not to take on the responsibility of handling secret tokens as mybinder.org is a public service and proof of technological concept.
```